apps/opik-backend/src/main/java/com/comet/opik/domain/TraceDAO.java

```diff
@@ -695,4 +708,18 @@ public Mono<List<WorkspaceAndResourceId>> getTraceWorkspace(
                 .collectList();
     }
 
+    @Override
+    @com.newrelic.api.agent.Trace(dispatcher = true)
+    public Mono<TraceCntResponse> cntTracesPerWorkspace(Connection connection) {
+
+        var statement = connection.createStatement(TRACE_CNT_BY_WORKSPACE_ID);
+
+        return Mono.from(statement.execute())
+                .flatMapMany(result -> result.map((row, rowMetadata) -> new TraceCntResponse.WorkspaceTracesCnt(
+                        row.get("workspace_id", String.class),
+                        row.get("traces_count", Integer.class))))
+                .collectList()
+                .flatMap(items -> Mono.just(new TraceCntResponse(items)));
```

Comment:
The DAO should probably better return the Flux of items and then the service should be the right place to collect them into the response.

Reply:
Done